src/main/java/frc/robot/subsystems/PivotSubsystem.java

```diff
@@ -1,144 +1,137 @@
 import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
-
-import com.revrobotics.CANSparkFlex;
 import edu.wpi.first.wpilibj2.command.Command;
-import edu.wpi.first.wpilibj2.command.InstantCommand;
-import edu.wpi.first.wpilibj2.command.ProfiledPIDSubsystem;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
 import frc.robot.Constants.Constants.PivotConstants;
 
 
+import static frc.robot.Constants.Constants.PivotConstants.pivotEncoderChannel;
+
+public class PivotSubsystem extends SubsystemBase {
+
+    //makes motors
+    public final CANSparkFlex pivotMotor1 = new CANSparkFlex(PivotConstants.pivotMotor1ID, CANSparkLowLevel.MotorType.kBrushless);
+    public final CANSparkFlex pivotMotor2 = new CANSparkFlex(PivotConstants.pivotMotor2ID, CANSparkLowLevel.MotorType.kBrushless);
+
+    //gets encoders
+    public final RelativeEncoder pivotEncoder = pivotMotor1.getEncoder();//change later
 
-public class PivotSubsystem extends ProfiledPIDSubsystem {
-    private final CANSparkFlex pivotMotor1 = new CANSparkFlex(PivotConstants.pivotMotor1ID, CANSparkLowLevel.MotorType.kBrushless);
-    private final CANSparkFlex pivotMotor2 = new CANSparkFlex(PivotConstants.pivotMotor2ID, CANSparkLowLevel.MotorType.kBrushless);
-    private final RelativeEncoder pivotEncoder = pivotMotor1.getEncoder();
-    private final ArmFeedforward pivotFeedForward =
-            new ArmFeedforward(
-                    1,1.26,0.15,0.02);  //kSVolts, kGVolts, kVVoltSecondPerRad, kAVoltSecondSquaredPerRad
+
+    //makes PID for motors
+    private final SparkPIDController pivotPIDController = pivotMotor1.getPIDController();
+
+    //motion profiling
+    private final TrapezoidProfile pivot1Profile = new TrapezoidProfile(new TrapezoidProfile.Constraints(3.0, 1.0));
+    private final TrapezoidProfile pivot2Profile = new TrapezoidProfile(new TrapezoidProfile.Constraints(3.0, 1.0));
+    TrapezoidProfile.State motor1Setpoint = new TrapezoidProfile.State();
+    TrapezoidProfile.State motor2Setpoint = new TrapezoidProfile.State();
+    TrapezoidProfile.State motor1Goal = new TrapezoidProfile.State();
+    TrapezoidProfile.State motor2Goal = new TrapezoidProfile.State();
+
+    private double currentPosition = 0;
 
     public PivotSubsystem(){
-        super(
-                new ProfiledPIDController(
-                        0.5,
-                        0,
-                        0.32,
-                        new TrapezoidProfile.Constraints(
-                                0, //kMaxVelocityRadPerSecond,
-                                0 //kMaxAccelerationRadPerSecSquared)),
-                )));
-
-        pivotEncoder.setPositionConversionFactor(1.0/27);
-        pivotMotor1.setSmartCurrentLimit(40);
-        pivotMotor2.setSmartCurrentLimit(40);
-        pivotMotor1.setSoftLimit(CANSparkBase.SoftLimitDirection.kForward,180); //TODO tune
```

Comment:
This should be tuned by now

Reply:
we are planning to use the absolute encoder, no? that can be done later